tests/cli/test_cli.py

```diff
@@ -0,0 +1,505 @@
+    assert "-ORDER BY" not in result.output
+    assert "Directly Modified: sqlmesh_example__dev.full_model (Breaking)" not in result.output
+    # only incremental_model backfilled
+    assert "sqlmesh_example__dev.incremental_model evaluated in" in result.output
+    assert "sqlmesh_example__dev.full_model evaluated in" not in result.output
+    assert_backfill_success(result)
+
+
+def test_plan_dev_backfill(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # Create and backfill `prod` environment
+    runner.invoke(cli, ["--paths", tmp_path, "plan", "--auto-apply"])
+
+    update_incremental_model(tmp_path)
+    update_full_model(tmp_path)
+
+    # full_model change makes test fail, so we skip it
+    result = runner.invoke(
+        cli,
+        [
+            "--paths",
+            tmp_path,
+            "plan",
+            "dev",
+            "--skip-tests",
+            "--backfill-model",
+            "sqlmesh_example.incremental_model",
+        ],
+        input="\n\ny\n",
+    )
+    assert result.exit_code == 0
+    assert_new_env(result, "dev")
+    # both model diffs present
+    assert "-ORDER BY" in result.output
+    assert "Directly Modified: sqlmesh_example__dev.full_model (Breaking)" in result.output
+    assert "+  'a' AS new_col" in result.output
+    assert (
+        "Directly Modified: sqlmesh_example__dev.incremental_model (Non-breaking)" in result.output
+    )
+    # only incremental_model backfilled
+    assert "sqlmesh_example__dev.incremental_model evaluated in" in result.output
+    assert "sqlmesh_example__dev.full_model evaluated in" not in result.output
+    assert_backfill_success(result)
+
+
+def test_run_no_prod(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # Error if no env specified and `prod` doesn't exist
+    result = runner.invoke(cli, ["--paths", tmp_path, "run"])
+    assert result.exit_code == 1
+    assert "Error: Environment 'prod' was not found." in result.output
+
+
+@freeze_time(FREEZE_TIME)
+def test_run_dev(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # Create dev environment but DO NOT backfill
+    runner.invoke(cli, ["--paths", tmp_path, "plan", "dev", "--skip-backfill"], input="y\n")
+
+    # Confirm backfill occurs when we run non-backfilled dev env
+    result = runner.invoke(cli, ["--paths", tmp_path, "run", "dev"])
+    assert result.exit_code == 0
+    assert_model_batches_executed(result)
+
+
+@freeze_time(FREEZE_TIME)
+def test_run_cron_not_elapsed(runner, tmp_path):
+    create_example_project(tmp_path)
+
+    # Create and backfill `prod` environment
+    runner.invoke(cli, ["--paths", tmp_path, "plan", "--no-prompts", "--auto-apply"])
+
+    # No error and no output if `prod` environment exists and cron has not elapsed
+    result = runner.invoke(cli, ["--paths", tmp_path, "run"])
+    assert result.exit_code == 0
+
+    # INFO messages are printed to console when tests run in some environments.
```

Comment:
What do you mean by some environments here? Btw pytest has a `caplog` fixture that can be used to check what is logged during a test. Not sure if that would be helpful here.

Reply:
Yeah, it's weird. 

The test was passing with `output == ""` when I ran the test from VSCode or with `pytest <file name>` in CLI, but when I ran `make fast-test` the output contained those info messages and test failed.

I assumed that we can't fully control the behavior of info messages across all places these tests will run, so this version of the assert handles output with or without them.